common/src/state/mod.rs

```diff
@@ -436,8 +438,17 @@ impl State {
         match event {
             MessageEvent::Received {
                 conversation_id,
-                message,
+                mut message,
             } => {
+                if message.inner.lines().iter().any(|s| s.contains('@')) {
+                    if let Some(ids) = self
+                        .get_chat_by_id(conversation_id)
+                        .map(|c| self.chat_participants(&c))
+                    {
+                        message.resolve_message(&ids, &self.get_own_identity().did_key());
+                    }
+                }
```

Comment:
I think it better if the check for the `@` character was moved into the `resolve_message()` function. I think the function could be named better too. recommend something like `insert_did()`